pkg/config/nodetreemodel/read_config_file.go

```diff
@@ -19,7 +20,12 @@ import (
 
 func (c *ntmConfig) getConfigFile() string {
 	if c.configFile == "" {
-		return "datadog.yaml"
+		for _, path := range c.configPaths {
+			configFilePath := filepath.Join(path, c.configName+".yaml")
+			if _, err := os.Stat(configFilePath); err == nil {
+				return configFilePath
```

Comment:
How about assigning to `c.configFile` here and breaking out of the loop. That way the caller doesn't need to do it.